### PR TITLE
Add protobuf language support

### PR DESCRIPTION
**Pull request by TitanNano**
Adds basic syntax highlighting for protobuf (*.proto) files.

- [x] Added an entry to `CHANGELOG.md` if this change could be valuable to users

### CODE REVIEW

File: lapce-core/src/language.rs [hunk 1]
```diff
@@ -641,6 +643,18 @@ const LANGUAGES: &[SyntaxProperties] = &[
         sticky_headers: &[],
         extensions: &["prisma"],
     },
+    #[cfg(feature = "lang-proto")]
```
Comment (panekj) [inline lapce-core/src/language.rs:646]:
```suggestion
    #[cfg(feature = "lang-protobuf")]
```

File: lapce-core/src/syntax/highlight.rs [hunk 2]
```diff
@@ -85,6 +85,7 @@ declare_language_highlights!(
     OcamlInterface: "lang-ocaml",
     Php: "lang-php",
     Prisma: "lang-prisma",
+    ProtoBuf: "lang-proto",
```
Comment (panekj) [inline lapce-core/src/syntax/highlight.rs:88]:
```suggestion
    ProtoBuf: "lang-protobuf",
```

File: lapce-ui/Cargo.toml [hunk 3]
```diff
@@ -79,6 +79,7 @@ all-languages = [
     "lang-ocaml",
     "lang-php",
     "lang-prisma",
+    "lang-proto",
```
Comment (panekj) [inline lapce-ui/Cargo.toml:82]:
```suggestion
    "lang-protobuf",
```

File: lapce-ui/Cargo.toml [hunk 4]
```diff
@@ -128,6 +129,7 @@ lang-nix = ["lapce-core/lang-nix"]
 lang-ocaml = ["lapce-core/lang-ocaml"]
 lang-php = ["lapce-core/lang-php"]
 lang-prisma = ["lapce-core/lang-prisma"]
+lang-proto = ["lapce-core/lang-proto"]
```
Comment (panekj) [inline lapce-ui/Cargo.toml:132]:
```suggestion
lang-protobuf = ["lapce-core/lang-protobuf"]
```

File: lapce-core/src/language.rs [hunk 5]
```diff
@@ -176,6 +176,8 @@ pub enum LapceLanguage {
     Php,
     #[cfg(feature = "lang-prisma")]
     Prisma,
+    #[cfg(feature = "lang-proto")]
```
Comment (panekj) [inline lapce-core/src/language.rs:179]:
```suggestion
    #[cfg(feature = "lang-protobuf")]
```

File: lapce-core/src/language.rs [hunk 6]
```diff
@@ -641,6 +643,18 @@ const LANGUAGES: &[SyntaxProperties] = &[
         sticky_headers: &[],
         extensions: &["prisma"],
     },
+    #[cfg(feature = "lang-proto")]
+    SyntaxProperties {
+        id: LapceLanguage::ProtoBuf,
+        language: tree_sitter_protobuf::language,
+        highlight: include_str!("../queries/proto/highlights.scm"),
```
Comment (panekj) [inline lapce-core/src/language.rs:650]:
```suggestion
        highlight: include_str!("../queries/protobuf/highlights.scm"),
```

File: lapce-core/Cargo.toml [hunk 7]
```diff
@@ -112,6 +113,7 @@ lang-nix = ["dep:tree-sitter-nix"]
 lang-ocaml = ["dep:tree-sitter-ocaml"]
 lang-php = ["dep:tree-sitter-php"]
 lang-prisma = ["dep:tree-sitter-prisma-io"]
+lang-proto = ["dep:tree-sitter-protobuf"]
```
Comment (panekj) [inline lapce-core/Cargo.toml:116]:
```suggestion
lang-protobuf = ["dep:tree-sitter-protobuf"]
```

**Review by panekj — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -13,7 +14,6 @@
 - [#1702](https://github.com/lapce/lapce/pull/1702): Improved svelte treesitter queries
 - [#1690](https://github.com/lapce/lapce/pull/1690): Add codelens and sticky headers for Dart
 - [#1711](https://github.com/lapce/lapce/pull/1711): Add zstd support for plugin
-- [#1715](https://github.com/lapce/lapce/pull/1715): Add support for requests from plugins
```
Comment (panekj) [inline CHANGELOG.md:16]: mistake?

Comment (TitanNano) [reply CHANGELOG.md:16]: yes :thinking:

File: lapce-core/queries/protobuf/highlights.scm [hunk 9]
```diff
@@ -0,0 +1,58 @@
+[
+  "syntax"
+  "package"
+  "option"
+  "import"
+  "service"
```
Comment (panekj) [inline lapce-core/queries/protobuf/highlights.scm:6]: Where are this files from?

Comment (TitanNano) [reply lapce-core/queries/protobuf/highlights.scm:6]: `folds.scm` is from the tree-sitter-protobuf repo and `highlights.scm` comes from helix-editor.

Comment (panekj) [reply lapce-core/queries/protobuf/highlights.scm:6]: could you please include that information just like other queries have?

**Review by panekj — APPROVED** [review]

**Pull request merged**